### PR TITLE
[AMS-566] unify link and videoLink constraints

### PR DESCRIPTION
**Pull request by xakep139**
Additionally preprocess links

### CODE REVIEW

File: src/VStore/Objects/ContentPreprocessing/ElementTextHarmonizer.cs [hunk 1]
```diff
@@ -20,6 +20,13 @@ public static class ElementTextHarmonizer
                               .Replace((char)160, ' ')
                               .TrimEnd();
 
+        public static string ProcessLink(string rawLink) =>
+            string.IsNullOrEmpty(rawLink)
+                ? rawLink
+                : rawLink.Replace("\r\n", "\n")
+                         .Replace((char)160, ' ')
+                         .Trim();
```
Comment (xakep139) [inline src/VStore/Objects/ContentPreprocessing/ElementTextHarmonizer.cs]: Trim links on both sides

File: src/VStore/Objects/ContentValidation/TextValidationUtils.cs [hunk 2]
```diff
@@ -19,10 +19,10 @@ public static IReadOnlyCollection<string> GetTooLongWords(string text, int maxSy
             return
                 text.Split(separators, StringSplitOptions.RemoveEmptyEntries)
                     .Where(x => x.Length > maxSymbolsPerWord)
-                    .ToArray();
+                    .ToList();
```
Comment (xakep139) [inline src/VStore/Objects/ContentValidation/TextValidationUtils.cs]: [AMS-173]

File: src/VStore/Objects/ContentPreprocessing/ElementTextHarmonizer.cs (hunk 1 above)
Comment (denisivan0v) [inline src/VStore/Objects/ContentPreprocessing/ElementTextHarmonizer.cs]: Is `\n` symbol valid for links?

Comment (xakep139) [inline src/VStore/Objects/ContentPreprocessing/ElementTextHarmonizer.cs]: It was taken from ERM preprocessing. `'n` is valid only as leading or closing whitespace symbol

File: src/VStore/Descriptors/Templates/LinkElementConstraints.cs [hunk 3]
```diff
@@ -2,18 +2,43 @@
 
 namespace NuClear.VStore.Descriptors.Templates
 {
-    public class LinkElementConstraints : PlainTextElementConstraints, ILinkElementConstraints, IEquatable<LinkElementConstraints>
+    public class LinkElementConstraints : ILinkElementConstraints, ITextElementConstraints, IEquatable<LinkElementConstraints>
     {
         public bool ValidLink => true;
+        public int? MaxSymbols { get; set; }
+        public bool WithoutControlChars => true;
+        public bool WithoutNonBreakingSpace => true;
```
Comment (xakep139) [inline src/VStore/Descriptors/Templates/LinkElementConstraints.cs]: Discussed

File: src/VStore/Objects/ContentPreprocessing/ElementTextHarmonizer.cs (hunk 1 above)
Comment (xakep139) [inline src/VStore/Objects/ContentPreprocessing/ElementTextHarmonizer.cs]: Redefined in https://github.com/2gis/nuclear-vstore/pull/78/commits/298fa5acc94dbcfb30048b02d652a7f08df48845

**Pull request merged**